manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -100,10 +100,10 @@ func (tr *TaskReaper) Run(ctx context.Context) {
 			// Serviceless tasks can be cleaned up right away since they are not attached to a service.
 			tr.cleanup = append(tr.cleanup, t.ID)
 		}
-		// tasks with desired state REMOVE that have progressed beyond COMPLETE can be cleaned up
-		// right away
+		// tasks with desired state REMOVE that have progressed beyond COMPLETE or
+		// haven't been assigned yet can be cleaned up right away
 		for _, t := range removeTasks {
-			if t.Status.State >= api.TaskStateCompleted {
+			if t.Status.State >= api.TaskStateCompleted || t.Status.State < api.TaskStateAssigned {
```

Comment:
nit: I'd suggest making it:

t.Status.State < api.TaskStateAssigned || t.Status.State >= api.TaskStateCompleted

slightly better for reading